lib/multi_fetch_fragments.rb

```diff
@@ -37,6 +37,13 @@ def render_collection_with_multi_fetch_cache
           expanded_key = @view.controller.fragment_cache_key(key_with_optional_digest) 
 
           keys_to_collection_map[expanded_key] = item
+
+          if item.respond_to?(:cache_life)
```

Comment:
In your other fix, the API always responds with `1.year` or a smaller `Time` object. What is the purpose of also adding default value logic here?

If this is necessary, suggest writing in a one line statement:

```ruby
@expires_collection[expanded_key] = item.cache_life || 1.year
```